paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonScan.scala

```diff
@@ -36,35 +36,53 @@ case class PaimonScan(
     requiredSchema: StructType,
     filters: Seq[Predicate],
     reservedFilters: Seq[Filter],
-    pushDownLimit: Option[Int])
+    pushDownLimit: Option[Int],
+    disableBucketedScan: Boolean = false)
   extends PaimonBaseScan(table, requiredSchema, filters, reservedFilters, pushDownLimit)
   with SupportsRuntimeFiltering
   with SupportsReportPartitioning {
 
-  override def outputPartitioning(): Partitioning = {
+  def withDisabledBucketedScan(): PaimonScan = {
```

Comment:
disableBucketedScan

Reply:
good naming! but it seems conflict with line40..

Reply:
bucketedScanDisabled and disableBucketedScan?

Reply:
+1, addressed